controller/pods.go

```diff
@@ -223,15 +243,40 @@ func (c *Controller) podUpdated(oldObj, newObj interface{}) {
 		np := cniPodNetworkProperty{}
 		for _, n := range nets {
 			np.IfMAC = n.IfMAC
+			np.IsPrimary = n.IsPrimary
 			err := c.addNetwork(newPod, n.NetworkName, np)
 			if err != nil {
-				glog.V(4).Infof("Failed to add network %s on pod %s", n.NetworkName, podName)
+				glog.Errorf("Failed to add network %s on pod %s", n.NetworkName, podName)
+			}
+		}
+	}
+}
+
+func (c *Controller) eventQueueWorker() {
+	for {
+		c.eventQueue.cond.L.Lock()
+
+		for c.eventQueue.q.Len() == 0 {
+			c.eventQueue.cond.Wait()
+		}
+
+		for {
+			ev := c.eventQueue.Dequeue()
+			if ev == nil {
+				break
 			}
+			glog.V(5).Infof("Processing event:", ev)
+			c.Process(ev)
 		}
+		c.eventQueue.cond.L.Unlock()
```

Comment:
You can not hold the lock while you are retrying for possibly 10 minutes, preventing adding ro removing any event in the queue.